source/extensions/filters/http/cache/cache_header_utility.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/filters/http/cache/cache_header_utility.h"
+
+#include <vector>
+
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/header_utility.h"
+#include "common/http/headers.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+namespace {
+const absl::string_view bytes_{"bytes"};
+}
+
+std::vector<RawByteRange> CacheHeaderUtility::getRanges(const Http::HeaderMap& request_headers) {
+  if (request_headers.Method() == nullptr ||
+      request_headers.Method()->value().getStringView() != Http::Headers::get().MethodValues.Get) {
+    // Range headers are only valid on GET requests so don't bother parsing
+    // range if we're not going to use it.
+    return {};
+  }
+
+  // Multiple instances of range/range-unit headers are considered invalid.
+  // https://tools.ietf.org/html/rfc7230#section-3.2.2
+  // TODO: should we log when encountering multiple headers?
```

Comment:
My guess would be no, although maybe a debug or trace level log would be appropriate.